pygments/lexers/elpi.py

```diff
@@ -0,0 +1,143 @@
+"""
+    pygments.lexers.elpi
+    ~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for the `Elpi <http://github.com/LPCIC/elpi>`_ programming language.
+
+    :copyright: Copyright 2006-2021 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, bygroups, include
+from pygments.token import \
+    Text, Comment, Operator, Keyword, Name, String, Number
+
+__all__ = ['ElpiLexer']
+
+class ElpiLexer(RegexLexer):
+    """
+    Lexer for the `Elpi <http://github.com/LPCIC/elpi>`_ programming language.
+
+    .. versionadded::2.11
+    """
+
+    name = 'Elpi'
+    aliases = ['elpi']
+    filenames = ['*.elpi']
+    mimetypes = ['text/x-elpi']
+    
+    lcase_re = r"[a-z]"
+    ucase_re = r"[A-Z]"
+    digit_re = r"[0-9]"
+    schar2_re = r"(\+|\*|/|\^|<|>|`|'|\?|@|#|~|=|&|!)"
+    schar_re = r"({}|-|\$|_)".format(schar2_re)
+    idchar_re = r"({}|{}|{}|{})".format(lcase_re,ucase_re,digit_re,schar_re)
```

Comment:
Instead of merging three separate rules here, `[\w\-$_]` would also do the trick and is much easier to parse for me. Similarly, I'd prefer `[<>+...]` instead of `+|<|>|...` as it reduces the "noise" while reading.

Reply:
I've addressed the second issue. The former, I'm not too happy, since I have this grammar in 3 places already (here, the actual parser, and vscode) and in all 3 I use these non-terminals (schar, schar2, ...). So I'd rather keep them in sync.

If this is blocking, I'll amend. Otherwise bare with me.